server/raft.go

```diff
@@ -4138,11 +4138,10 @@ func (n *raft) processVoteRequest(vr *voteRequest) error {
 		n.vote = vr.candidate
 		n.writeTermVote()
 		n.resetElectionTimeout()
-	} else {
-		if vr.term >= n.term && n.vote == noVote {
```

Comment:
Have removed the `vr.term >= n.term` part of the condition as well as the `n.term = vr.term` as they don't do anything at the moment.

If `vr.term < n.term` we already deny requests, and if `vr.term > n.term` we update above. So when we come here `vr.term == n.term` already.